stingray/modeling/gpmodeling.py

```diff
@@ -0,0 +1,850 @@
+    """A constant mean shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Constant amplitude of the flare.
+
+    Returns
+    -------
+    The constant value.
+    """
+    return mean_params["A"] * jnp.ones_like(t)
+
+
+def _skew_gaussian(t, mean_params):
+    """A skew gaussian flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    sig1:
+        The width parameter for the rising edge.
+    sig2:
+        The width parameter for the falling edge.
+
+    Returns
+    -------
+    The y values for skew gaussian flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    sig1 = jnp.atleast_1d(mean_params["sig1"])[:, jnp.newaxis]
+    sig2 = jnp.atleast_1d(mean_params["sig2"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A
+        * jnp.where(
+            t > t0,
+            jnp.exp(-((t - t0) ** 2) / (2 * (sig2**2))),
+            jnp.exp(-((t - t0) ** 2) / (2 * (sig1**2))),
+        ),
+        axis=0,
+    )
+
+
+def _skew_exponential(t, mean_params):
+    """A skew exponential flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    sig1:
+        The width parameter for the rising edge.
+    sig2:
+        The width parameter for the falling edge.
+
+    Returns
+    -------
+    The y values for exponential flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    sig1 = jnp.atleast_1d(mean_params["sig1"])[:, jnp.newaxis]
+    sig2 = jnp.atleast_1d(mean_params["sig2"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A
+        * jnp.where(
+            t > t0,
+            jnp.exp(-(t - t0) / (2 * (sig2**2))),
+            jnp.exp((t - t0) / (2 * (sig1**2))),
+        ),
+        axis=0,
+    )
+
+
+def _fred(t, mean_params):
+    """A fast rise exponential decay (FRED) flare shape.
+
+    Parameters
+    ----------
+    t:  jnp.ndarray
+        The time coordinates.
+    A:  jnp.int
+        Amplitude of the flare.
+    t0:
+        The location of the maximum.
+    phi:
+        Symmetry parameter of the flare.
+    delta:
+        Offset parameter of the flare.
+
+    Returns
+    -------
+    The y values for exponential flare.
+    """
+    A = jnp.atleast_1d(mean_params["A"])[:, jnp.newaxis]
+    t0 = jnp.atleast_1d(mean_params["t0"])[:, jnp.newaxis]
+    phi = jnp.atleast_1d(mean_params["phi"])[:, jnp.newaxis]
+    delta = jnp.atleast_1d(mean_params["delta"])[:, jnp.newaxis]
+
+    return jnp.sum(
+        A * jnp.exp(-phi * ((t + delta) / t0 + t0 / (t + delta))) * jnp.exp(2 * phi), axis=0
+    )
```

Comment:
same here, nicer to `return y` and define `y` on a previous line

Reply:
Done now